server/plugin/taskLinkPreview_test.go

```diff
@@ -1,58 +1,120 @@
 	mockedClient := mocks.NewMockClient(mockCtrl)
-	p.Client = mockedClient
+	p := setupMockPlugin(mockAPI, nil, mockedClient)
 	for _, testCase := range []struct {
 		description string
 		linkData    []string
+		err         error
 	}{
 		{
 			description: "PostTaskPreview: valid",
 			linkData:    []string{"https:", "", "dev.azure.com", "abc", "xyz", "_workitems", "edit", "1"},
 		},
+		{
+			description: "PostTaskPreview: invalid",
+			linkData:    []string{"https:", "", "dev.azure.com", "abc", "xyz", "_workitems", "edit", "1"},
+			err:         errors.New("mockError"),
+		},
 	} {
 		t.Run(testCase.description, func(t *testing.T) {
-			mockedClient.EXPECT().GetTask(gomock.Any(), gomock.Any(), gomock.Any(), "mockUserID").Return(&serializers.TaskValue{}, http.StatusOK, nil)
-			resp, err := p.PostTaskPreview(testCase.linkData, "mockUserID", "mockChannelID")
-			assert.Equal(t, "", err)
-			assert.NotNil(t, resp)
+			mockAPI.On("LogDebug", testutils.GetMockArgumentsWithType("string", 3)...)
+
+			mockedClient.EXPECT().GetTask(gomock.Any(), gomock.Any(), gomock.Any(), "mockUserID").Return(&serializers.TaskValue{}, http.StatusOK, testCase.err)
+			resp, msg := p.PostTaskPreview(testCase.linkData, "mockUserID", "mockChannelID")
+			assert.Equal(t, "", msg)
+			if testCase.err != nil {
+				assert.Nil(t, resp)
+			} else {
+				assert.NotNil(t, resp)
+			}
 		})
 	}
 }
 
 func TestPostPullRequestPreview(t *testing.T) {
-	p := Plugin{}
+	mockAPI := &plugintest.API{}
 	mockCtrl := gomock.NewController(t)
 	mockedClient := mocks.NewMockClient(mockCtrl)
-	p.Client = mockedClient
+	p := setupMockPlugin(mockAPI, nil, mockedClient)
 	for _, testCase := range []struct {
 		description string
 		linkData    []string
+		err         error
 	}{
 		{
 			description: "PostPullRequestPreview: valid",
 			linkData:    []string{"https:", "", "dev.azure.com", "abc", "xyz", "_git", "xyz", "pullrequest", "1"},
 		},
+		{
+			description: "PostPullRequestPreview: invalid",
+			linkData:    []string{"https:", "", "dev.azure.com", "abc", "xyz", "_git", "xyz", "pullrequest", "1"},
+			err:         errors.New("mockError"),
+		},
 	} {
 		t.Run(testCase.description, func(t *testing.T) {
-			mockedClient.EXPECT().GetPullRequest(gomock.Any(), gomock.Any(), gomock.Any(), "mockUserID").Return(&serializers.PullRequest{}, http.StatusOK, nil)
-			resp, stringErr := p.PostPullRequestPreview(testCase.linkData, "mockPullRequestLink", "mockUserID", "mockChannelID")
-			assert.Equal(t, "", stringErr)
-			assert.NotNil(t, resp)
+			mockAPI.On("LogDebug", testutils.GetMockArgumentsWithType("string", 3)...)
+
+			mockedClient.EXPECT().GetPullRequest(gomock.Any(), gomock.Any(), gomock.Any(), "mockUserID").Return(&serializers.PullRequest{}, http.StatusOK, testCase.err)
+			resp, msg := p.PostPullRequestPreview(testCase.linkData, "mockPullRequestLink", "mockUserID", "mockChannelID")
+			assert.Equal(t, "", msg)
+			if testCase.err != nil {
+				assert.Nil(t, resp)
```

Comment:
check error value if possible

Reply:
We are not getting an error in the response. We are just getting the `resp` which has different values which cannot be checked. And the `msg` is always an empty string.